cmd/packages.go

```diff
@@ -236,54 +235,66 @@ func isVerbose() (result bool) {
 	return appConfig.CliOptions.Verbosity > 0 || isPipedInput
 }
 
+func generateSBOM(userInput string, errs chan error) (*sbom.SBOM, *source.Source, error) {
+	tasks, err := tasks()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	src, cleanup, err := source.New(userInput, appConfig.Registry.ToOptions(), appConfig.Exclusions)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to construct source from user input %q: %w", userInput, err)
+	}
+	if cleanup != nil {
+		defer cleanup()
+	}
+
+	s := sbom.SBOM{
+		Source: src.Metadata,
+		Descriptor: sbom.Descriptor{
+			Name:          internal.ApplicationName,
+			Version:       version.FromBuild().Version,
+			Configuration: appConfig,
+		},
+	}
+
+	s = buildRelationships(s, src, tasks, errs)
+
+	return &s, src, nil
+}
+
+func buildRelationships(s sbom.SBOM, src *source.Source, tasks []task, errs chan error) sbom.SBOM {
+	var relationships []<-chan artifact.Relationship
+	for _, task := range tasks {
+		c := make(chan artifact.Relationship)
+		relationships = append(relationships, c)
+		go runTask(task, &s.Artifacts, src, c, errs)
+	}
+
+	s.Relationships = append(s.Relationships, mergeRelationships(relationships...)...)
+
+	return s
+}
+
 func packagesExecWorker(userInput string, writer sbom.Writer) <-chan error {
 	errs := make(chan error)
 	go func() {
 		defer close(errs)
-
-		tasks, err := tasks()
+		s, src, err := generateSBOM(userInput, errs)
```

Comment:
should have a `nil` check on `s` to ensure that an SBOM is provided or else error out (possible panic here)

Reply:
Checking now for nill sbom no panic